### DISCUSSION TITLE
Why is terminfo needed on a remote server when using SSH?

### DISCUSSION THREAD

**Discussion by samholmes** [kovidgoyal/kitty#5283] [Q&A] [answered]
I'd like to better understand why this `.terminfo` directory is needed when using SSH in kitty. Maybe if I understand the problem, I could better understand the solution (the Shell Integration).

I'd like know what effect the `kitty +kitten ssh` has on my server's state, and how to easily uninstall the shell integration if desired (leaving no artifacts behind). Does it effect some system-wide state, or does it only write to the `/home/<user>/.terminfo` directory of the user that has connected over SSH?

**Comment by kovidgoyal** [comment, marked as answer]
Google is your friend. In brief terminfo is needed because there is a
horrible piece of software called ncurses that a lot of older terminal
programs use without which they get things like colors and key handling
wrong.

And given that you are sshing in as a user the ssh kitten can only write
to those directories that user has write permission for. And it makes no
changes to server state whatsoever other than copying a few files over
that dont affect anything on the server, since they are all only used if
the correct environment variables are set.
